### PR TITLE
fixes to multiteacher, batching

### PR DESCRIPTION
**Pull request by alexholdenmiller**

### CODE REVIEW

File: parlai/core/worlds.py [hunk 1]
```diff
@@ -374,9 +373,10 @@ def parley_init(self):
                 while keep_looking:
                     self.world_idx = (self.world_idx + 1) % len(self.worlds)
                     keep_looking = (self.worlds[self.world_idx].epoch_done() and
-                                    start_idx != self.world_idx)
-                if start_idx == self.world_idx:
-                    return {'text': 'There are no more examples remaining.'}
+                                    self.world_idx != start_idx)
+                if self.worlds[self.world_idx].epoch_done():
```
Comment (jaseweston) [inline parlai/core/worlds.py]: cut

File: parlai/core/dict.py [hunk 2]
```diff
@@ -177,6 +191,9 @@ def __setitem__(self, key, value):
             self.tok2ind[key] = index
             self.ind2tok[index] = key
 
+    def freqs(self):
```
Comment (alexholdenmiller) [inline parlai/core/dict.py]: did it!

File: parlai/core/agents.py [hunk 3]
```diff
@@ -333,3 +344,14 @@ def report(self):
         if num_tasks > 0:
             m['accuracy'] = sum_accuracy / num_tasks
         return m
+
+    def reset(self):
```
Comment (alexholdenmiller) [inline parlai/core/agents.py]: will do

File: parlai/core/agents.py [hunk 4]
```diff
@@ -258,16 +258,20 @@ def __init__(self, opt, shared=None):
...
+            for k in tasks:
+                k = k.strip()
+                if k:
+                    opt_singletask = copy.deepcopy(opt)
+                    opt_singletask['task'] = k
+                    self.tasks.extend(create_task_agent_from_taskname(
+                        opt_singletask))
         self.task_idx = -1
         self.new_task = True
+        self.ignore_reply = False
```
Comment (jaseweston) [inline parlai/core/agents.py]: i thought this was going?

Comment (alexholdenmiller) [inline parlai/core/agents.py]: oops yep, fixed

**Pull request merged**